test/routes/src/server/routes/t_launchpad.js

```diff
@@ -569,6 +570,214 @@ describe('The Launchpad API endpoint', () => {
+        nock(lp_api_url)
+          .get(lp_snap_path)
+          .reply(200, {
+            name: lp_snap_name,
+            builds_collection_link: `${lp_api_url}${lp_builds_path}`
+          });
+
+        // when getting builds list (via builds_collection_link)
+        nock(lp_api_url)
+          .get(lp_builds_path)
+          .query({
+            'ws.start': 0,
+            'ws.size': 10
+          })
+          .reply(200, {
+            total_size: 1,
+            start: 0,
+            entries: [
+              {
+                'self_link': `${lp_api_url}${lp_snap_path}/+build/12345`,
+                'date_started': '2016-12-06T10:40:17.751937+00:00',
+                'datebuilt': '2016-12-06T10:47:59.384231+00:00',
+                'duration': '0:07:41.632294',
+                'datecreated': '2016-12-06T10:30:29.285058+00:00',
+                'buildstate': 'Successfully built',
+                'resource_type_link': `${lp_api_url}/devel/#snap_build`,
+                'build_log_url': `${lp_api_url}${lp_snap_path}/+build/12345/+files/buildlog_snap_ubuntu_xenial_armhf_${lp_snap_name}_BUILDING.txt.gz`,
+                'arch_tag': 'armhf',
+              }
+            ]
+          });
+      });
+
+      afterEach(() => {
+        nock.cleanAll();
+      });
+
+      it('should return a 200 OK response', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(200, done);
+      });
+
+      it('should return a "success" status', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasStatus('success'))
+          .end(done);
+      });
+
+      it('should return body with "snap-builds-found" message', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasMessage('snap-builds-found'))
+          .end(done);
+      });
+
+      it('should return builds list as payload message', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .end((err, res) => {
+            expect(res.body.payload.message.length).toEqual(1);
+            done(err);
+          });
+      });
+
+    });
+
+    context('when builds fail to fetch', () => {
+
+      beforeEach(() => {
+        // when getting snap data from API (via self_link)
+        nock(lp_api_url)
+          .get(lp_snap_path)
+          .reply(200, {
+            name: lp_snap_name,
+            builds_collection_link: `${lp_api_url}${lp_builds_path}`
+          });
+
+        // when getting builds list (via builds_collection_link)
+        nock(lp_api_url)
+          .get(lp_builds_path)
+          .query({
+            'ws.start': 0,
+            'ws.size': 10
+          })
+          .reply(404, 'Not found');
```

Comment:
The structure of the route tests should generally follow the structure of the route itself, so I think it's only important that they test that errors returned by Launchpad are passed through properly, as you do here.  The details of how the Launchpad client behaves belong in the unit tests for the Launchpad client itself.